bundle/parallel_test.go

```diff
@@ -0,0 +1,67 @@
+package bundle
+
+import (
+	"context"
+	"testing"
+
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/stretchr/testify/require"
+)
+
+type addToContainer struct {
+	container *[]int
+	err       bool
+}
+
+func (m *addToContainer) Apply(ctx context.Context, b *Bundle) diag.Diagnostics {
+	if m.err {
+		return diag.Errorf("error")
+	}
+
+	c := *m.container
+	c = append(c, 1)
+	*m.container = c
+	return nil
+}
+
+func (m *addToContainer) Name() string {
+	return "addToContainer"
+}
+
+func TestParallelMutatorWork(t *testing.T) {
+	b := &Bundle{
+		Config: config.Root{},
+	}
+
+	container := []int{}
+	m1 := &addToContainer{container: &container}
```

Comment:
Can we make the mutators different here, to ensure that all mutators are run? You could do that my parameterising the int that's being appended to the array.